packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -361,12 +361,12 @@ private MethodBodyStatement GetPropertyInitializers(IReadOnlyList<ParameterProvi
             {
                 if (param.Name == _rawDataField?.Name.ToVariableName())
                 {
-                    methodBodyStatements.Add(Assign(new MemberExpression(null, _rawDataField.Name), param));
+                    methodBodyStatements.Add(_rawDataField.Assign(param).Terminate());
                     continue;
                 }
 
                 ValueExpression initializationValue = param;
-                var initializationStatement = Assign(new MemberExpression(null, param.Name.FirstCharToUpperCase()), initializationValue);
+                var initializationStatement = param.AsPropertyExpression.Assign(initializationValue).Terminate();
```

Comment:
well I do not quite like this part.
We should not just magically change the name and "assume" this is the property name.
Imagine that we have the default acronym mapping and some of the name may not change as simple as "make the first character to upper case" (maybe its first word is `ip`)
I think this part should work in this way:
we find the property from the model by this parameter name, like `paramProperty` and then we should be able to just use
`paramProperty.Assgin(initializationValue).Terminate()`